addons/monitoring/lambda/main.go

```diff
@@ -242,29 +243,36 @@ func checkCrons(db *sql.DB, sess *session.Session) (err error) {
 		if row.num_errors == 0 {
 			continue
 		}
-		log.Printf("*** %s job had errors (runs: %d, errors: %d), alerting! (errors %s)", row.name, row.num_occurences, row.num_errors, row.most_recent_error)
+		log.Printf("*** %s job had errors (runs: %d, errors: %d), alerting! (errors %s)", row.name, row.num_occurences, row.num_errors, row.most_recent_error.String)
 		if row.num_occurences == 1 {
-			sendSNSMessage(fmt.Sprintf("Fleet cron '%s' (last updated %s) raised errors during its last run:\n%s", row.name, row.updated_at.String(), row.errors), "cronJobFailure", sess)
+			sendSNSMessage(fmt.Sprintf("Fleet cron '%s' (last updated %s) raised errors during its last run:\n%s", row.name, row.updated_at.String(), row.most_recent_error.String), "cronJobFailure", sess)
 		} else {
-			sendSNSMessage(fmt.Sprintf("Fleet cron '%s' (last updated %s) raised errors in %d of the previous %d runs; the most recent is:\n%s", row.name, row.last_updated_at.String(), row.num_errors, row.num_occurences, row.most_recent_error), "cronJobFailure", sess)
+			sendSNSMessage(fmt.Sprintf("Fleet cron '%s' (last updated %s) raised errors in %d of the previous %d runs; the most recent is:\n%s", row.name, row.last_updated_at.String(), row.num_errors, row.num_occurences, row.most_recent_error.String), "cronJobFailure", sess)
 		}
 	}
 
 	return nil
 }
 
 func handler(ctx context.Context, name NullEvent) error {
+	awsConfig := aws.NewConfig()
+	awsConfig = awsConfig.WithRegion(options.AWSRegion)
+	if options.AwsEndpointUrl != "" {
+		awsConfig = awsConfig.WithEndpoint(options.AwsEndpointUrl)
+	}
 	sess := session.Must(session.NewSessionWithOptions(
 		session.Options{
 			SharedConfigState: session.SharedConfigEnable,
-			Config: aws.Config{
-				Region: &options.AWSRegion,
-			},
+			Config:            *awsConfig,
 		},
 	))
 
 	db, err := setupDB(sess)
-	defer db.Close()
+	defer func() {
+		if db != nil {
+			db.Close()
+		}
+	}()
```

Comment:
This was causing a panic when the DB setup errored, since there was no db to close.